Diagnostics/start-windows-build.ps1

```diff
@@ -0,0 +1,388 @@
+    )
+    $remoteCMD = "if [[ -d $RemoteDirectoryPath ]]; then rm -rf $RemoteDirectoryPath; fi; mkdir -p $RemoteDirectoryPath"
+    Start-SSHCommand -Server $REMOTE_LOG_SERVER -User $REMOTE_USER -Key $env:SSH_KEY -Command $remoteCMD
+}
+
+function New-RemoteSymlink {
+    Param(
+        [Parameter(Mandatory=$true)]
+        [string]$RemotePath,
+        [Parameter(Mandatory=$false)]
+        [string]$RemoteSymlinkPath
+    )
+    $remoteCMD = "if [[ -h $RemoteSymlinkPath ]]; then unlink $RemoteSymlinkPath; fi; ln -s $RemotePath $RemoteSymlinkPath"
+    Start-SSHCommand -Server $REMOTE_LOG_SERVER -User $REMOTE_USER -Key $env:SSH_KEY -Command $remoteCMD
+}
+function Get-DiagnosticsBuildRelativePath {
+    $repositoryOwner = $GitURL.Split("/")[-2]
+    $mesosCommitID = Get-LatestCommitID
+    return "$repositoryOwner/$Branch/$diagnosticsCommitID"
+}
+
+function Get-RemoteBuildDirectoryPath {
+    $relativePath = Get-DiagnosticsBuildRelativePath
+    return "$REMOTE_DIAGNOSTICS_BUILD_DIR/$relativePath"
+}
+
+function Get-BuildOutputsUrl {
+    $relativePath = Get-DiagnosticsBuildRelativePath
+    return "$DIAGNOSTICS_BUILD_BASE_URL/$relativePath"
+}
+
+function Get-BuildLogsUrl {
+    $buildOutUrl = Get-BuildOutputsUrl
+    return "$buildOutUrl/logs"
+}
+
+function Get-BuildBinariesUrl {
+    $buildOutUrl = Get-BuildOutputsUrl
+    return "$buildOutUrl/binaries"
+}
+
+function New-RemoteLatestSymlinks {
+    $remoteDirPath = Get-RemoteBuildDirectoryPath
+    $baseDir = (Split-Path -Path $remoteDirPath -Parent) -replace '\\', '/'
+    New-RemoteSymlink -RemotePath $remoteDirPath -RemoteSymlinkPath "$baseDir/latest"
+
+    $repoDir = (Split-Path -Path $baseDir -Parent) -replace '\\', '/'
+    New-RemoteSymlink -RemotePath $remoteDirPath -RemoteSymlinkPath "$repoDir/latest"
+}
+
+function Start-LogServerFilesUpload {
+    $consoleUrl = "${JENKINS_SERVER_URL}/job/${env:JOB_NAME}/${env:BUILD_NUMBER}/consoleText"
+    Start-FileDownload -Force -URL $consoleUrl -Destination "$DIAGNOSTICS_BUILD_LOGS_DIR\console-jenkins.log"
+    $remoteDirPath = Get-RemoteBuildDirectoryPath
+    New-RemoteDirectory -RemoteDirectoryPath $remoteDirPath
+    Copy-FilesToRemoteServer "$DIAGNOSTICS_BUILD_OUT_DIR\*" $remoteDirPath
+    $buildOutputsUrl = Get-BuildOutputsUrl
+    $global:PARAMETERS["BUILD_OUTPUTS_URL"] = $buildOutputsUrl
+    Write-Output "Build artifacts can be found at: $buildOutputsUrl"
+    if($global:PARAMETERS["BUILD_STATUS"] -eq "PASS") {
+        New-RemoteLatestSymlinks
+    }
+ }
+
+function Start-EnvironmentCleanup {
+    # Stop any potential hanging process
+    $processes = @('go')
+    $processes | Foreach-Object { Stop-Process -Name $_ -Force -ErrorAction SilentlyContinue }
+    cmd.exe /C "rmdir /s /q $DIAGNOSTICS_DIR > nul 2>&1"
+    Clear-Variable -Name "LATEST_COMMIT_ID"
+}
+
+function Get-SuccessBuildMessage {
+    return "Successful Diagnostics nightly build and testing for repository $GitURL on branch $Branch"
```

Comment:
Can you please remove the word "nightly" from the message ? We are planning to have this run every check-in. The way we do it right now for Mesos and Spartan.